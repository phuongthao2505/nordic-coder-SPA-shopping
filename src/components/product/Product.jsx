import React, { Component } from 'react';

export default class Product extends Component {
  handleCartClick = () => {
    if (typeof this.props.onAddToCart === 'function') {
      this.props.onAddToCart(this.props.id);
    }
  };
  handleDetail = (id) =>{
  };
  render() {
    const { id, name, image, image2, category, discountPrice, price, unitSold } = this.props;

    return (
      <div className="col-xl-4 col-lg-6 col-md-6">
        <div className="product-wrapper mb-50">
          <div className="product-img mb-25">
            <a className="image-link">
              <img src={image} alt={name} />
              <img className="secondary-img" src={image2} alt={name} />
            </a>
            <div className="product-action text-center">
              <a onClick={this.handleCartClick} title="Add to cart">
                <i className="fas fa-shopping-cart" />
              </a>
              <a title="View details" onClick={() => this.handleDetail(id)}>
                <i className="fas fa-search" />
              </a>
            </div>
          </div>
          <div className="product-content pr-0">
            <div className="pro-cat mb-10">
              <a>{category}</a>
            </div>
            <h4>
              <a>{name}</a>
            </h4>
            <div className="product-meta">
              <div className="pro-price">
                <span>${discountPrice} USD</span>
                <span className="old-price">${price} USD</span>
                <span className="unit-sold">{unitSold} sold</span>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
