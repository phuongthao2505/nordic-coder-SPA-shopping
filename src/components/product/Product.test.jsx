import React from 'react';
import renderer from 'react-test-renderer';

import Product from './Product';

const mockProduct = {
  id: '0001',
  name: 'Minimal Deco Furniture',
  image: './assets/pro1.jpg',
  image2: './assets/pro2.jpg',
  category: 'furniture',
  discountPrice: 119.0,
  price: 230.0,
  unitSold: 15,
};

test('Product renderer', () => {
  const component = renderer.create(<Product {...mockProduct} />);

  // expect(component.props).toBe()
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
