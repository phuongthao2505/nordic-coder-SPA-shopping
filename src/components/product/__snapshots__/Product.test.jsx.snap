// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product renderer 1`] = `
<div
  className="col-xl-4 col-lg-6 col-md-6"
>
  <div
    className="product-wrapper mb-50"
  >
    <div
      className="product-img mb-25"
    >
      <a
        className="image-link"
      >
        <img
          alt="Minimal Deco Furniture"
          src="./assets/pro1.jpg"
        />
        <img
          alt="Minimal Deco Furniture"
          className="secondary-img"
          src="./assets/pro2.jpg"
        />
      </a>
      <div
        className="product-action text-center"
      >
        <a
          onClick={[Function]}
          title="Add to cart"
        >
          <i
            className="fas fa-shopping-cart"
          />
        </a>
        <a
          title="View details"
        >
          <i
            className="fas fa-search"
          />
        </a>
      </div>
    </div>
    <div
      className="product-content pr-0"
    >
      <div
        className="pro-cat mb-10"
      >
        <a>
          furniture
        </a>
      </div>
      <h4>
        <a>
          Minimal Deco Furniture
        </a>
      </h4>
      <div
        className="product-meta"
      >
        <div
          className="pro-price"
        >
          <span>
            $
            119
             USD
          </span>
          <span
            className="old-price"
          >
            $
            230
             USD
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
